RUMM-883 Add support for 3rd party networking libraries (including Alamofire) (#340)
DataDog/dd-sdk-ios: Sources/DatadogExtensions/Alamofire/README.md

# Datadog Integration for Alamofire

`DatadogAlamofireExtension` enables `Alamofire.Session` auto instrumentation with Datadog SDK.

> It's a counterpart of `DDURLSessionDelegate` which we provide for native `URLSession` instrumentation.

Reviewer: ```suggestion
It's a counterpart of `DDURLSessionDelegate`, which is provided for native `URLSession` instrumentation.
```

> ## Getting Started

Reviewer: ```suggestion
## Getting started
```

### CocoaPods

To include the Datadog integration for [Alamofire][1] in your project, add the

> following to your `Podfile`.

Reviewer: ```suggestion
following to your `Podfile`:
```

```ruby
pod 'DatadogSDKAlamofireExtension'
```
`DatadogSDKAlamofireExtension` requires Datadog SDK `1.4.0` (or higher).

### Carthage and SPM

> Although our [Alamofire][1] integration doesn't currently support [Carthage][2] nor [SPM][3], the number of code needed to set it up is very low and you may want to just include the source files from this folder directly in your project.

Reviewer: ```suggestion
The Datadog [Alamofire][1] integration doesn't support [Carthage][2] or [SPM][3], however, the code needed for set up is very low. You may want to include the source files from this folder directly in your project.
```

### Initial setup

Follow the regular steps for initializing Datadog SDK for [Tracing][4] or [RUM][5].

Instead of using `DDURLSessionDelegate` for `URLSession`, use `DDEventMonitor` and `DDRequestInterceptor` for `Alamofire.Session`:

```swift
import DatadogAlamofireExtension
import Alamofire

let alamofireSession = Session(
   interceptor: DDRequestInterceptor(),
   eventMonitors: [DDEventMonitor()]
)
```

> This will make the Datadog SDK track requests sent from this instance of the `Alamofire.Session`.

Reviewer: ```suggestion
Using this setup sends the Datadog SDK track requests from this instance of the `Alamofire.Session`.
```
